### PR TITLE
Update United States holidays: Columbus Day adjustment

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

[Columbus Day](https://en.wikipedia.org/wiki/Columbus_Day) adjustment:
- Columbus Day with this exact name now exist only in 16 states and the territory of American Samoa
- some states have this holiday as ["Indigenous Peoples Day"](https://en.wikipedia.org/wiki/Indigenous_Peoples%27_Day_(United_States)) or other name

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by KJhellico** [comment]
@PPsyrius , @hissingpanda, I'd be glad to see your thoughts.

File: holidays/countries/united_states.py [hunk 1]
```diff
@@ -210,6 +233,11 @@ def _add_subdiv_ak_holidays(self):
             else:
                 self._add_holiday_mar_30(name)
 
+        # Indigenous Peoples Day
+        if self._year >= 1971:
+            name = "Indigenous Peoples Day" if self._year >= 2015 else "Columbus Day"
+            self._add_holiday_2nd_mon_of_oct(name)
```
Comment (arkid15r) [inline holidays/countries/united_states.py:239]:
```suggestion
            self._add_holiday_2nd_mon_of_oct(
                "Indigenous Peoples Day" if self._year >= 2015 else "Columbus Day"
            )

```

File: holidays/countries/united_states.py [hunk 2]
```diff
@@ -234,6 +262,15 @@ def _add_subdiv_al_holidays(self):
...
             self._add_holiday_1st_mon_of_jun("Jefferson Davis Birthday")
 
+        # Columbus Day / American Indian Heritage Day / Fraternal Day
+        if self._year >= 1971:
+            name = (
+                "Columbus Day / American Indian Heritage Day / Fraternal Day"
+                if self._year >= 2000
+                else "Columbus Day / Fraternal Day"
+            )
+            self._add_holiday_2nd_mon_of_oct(name)
```
Comment (arkid15r) [inline holidays/countries/united_states.py:272]:
```suggestion
            self._add_holiday_2nd_mon_of_oct(
                "Columbus Day / American Indian Heritage Day / Fraternal Day"
                if self._year >= 2000
                else "Columbus Day / Fraternal Day"
            )
```

well, here and after where name is going to be used only once

File: holidays/countries/united_states.py [hunk 3]
```diff
@@ -210,6 +233,11 @@ def _add_subdiv_ak_holidays(self):
             else:
                 self._add_holiday_mar_30(name)
 
+        # Indigenous Peoples Day
```
Comment (arkid15r) [inline holidays/countries/united_states.py:236]: Indigenous Peoples' Day
https://www.whitehouse.gov/briefing-room/presidential-actions/2021/10/08/a-proclamation-indigenous-peoples-day-2021/

**Review by arkid15r — COMMENTED** [review]
👍 

Some minor comments:

**Comment by PPsyrius** [comment]
Update: Seems like Minnesota (MN) just announced the change too yesterday (2023-10-09): https://www.wdio.com/front-page/top-stories/october-9th-is-officially-indigenous-peoples-day-in-minnesota/

File: tests/countries/test_united_states.py [hunk 4]
```diff
@@ -469,17 +513,95 @@ def test_columbus_day_states(self):
...
-            self.assertNoHolidayName(name, self.state_hols[subdiv])
-
-        self.assertHolidayName("Native American Day", self.state_hols["SD"], dt)
+        for subdiv, change_year in (
+            ("AK", 2015),
+            ("DC", 2019),
+            ("ME", 2019),
+            ("NM", 2019),
+            ("NE", 2020),
+            ("VA", 2020),
```
Comment (PPsyrius) [inline tests/countries/test_united_states.py:522]: Not **Vermont (VT)**?
https://www.washingtonpost.com/nation/2019/04/20/vermont-passes-bill-abolishing-columbus-day-favor-indigenous-peoples-day/, first started in 2019 too, not 2020

File: tests/countries/test_united_states.py [hunk 5]
```diff
@@ -469,17 +513,95 @@ def test_columbus_day_states(self):
...
+            "2021-10-11",
+            "2022-10-10",
+            "2023-10-09",
+        )
+        self.assertNoHolidayName(name_1, self.state_hols["AL"], range(1865, 1970))
+        self.assertHolidayName(name_1, self.state_hols["AL"], range(1971, 2000))
+        self.assertHolidayName(name_2, self.state_hols["AL"], range(2000, 2050))
+        self.assertHoliday(self.state_hols["AL"], dt)
+
+    def test_columbus_day_ri(self):
```
Comment (PPsyrius) [inline tests/countries/test_united_states.py:557]: Any source on Rhode Island ones?

File: tests/countries/test_united_states.py (hunk 4 above)
Comment (KJhellico) [reply tests/countries/test_united_states.py:522]: [Here](https://www.pewresearch.org/short-reads/2023/10/05/working-on-columbus-day-or-indigenous-peoples-day-it-depends-on-where-your-job-is/) says: 

> While the day is a legal holiday in Vermont, collective bargaining agreements allow state employees to use it as a floating holiday.

so I'm not sure if we should add this holiday for Vermont.

File: tests/countries/test_united_states.py (hunk 5 above)
Comment (KJhellico) [reply tests/countries/test_united_states.py:557]: https://web.archive.org/web/20230000000000*/https://www.sos.ri.gov/divisions/civics-and-education/reference-desk/ri-state-holidays

**Comment by KJhellico** [comment]
> Update: Seems like Minnesota (MN) just announced the change too yesterday (2023-10-09): https://www.wdio.com/front-page/top-stories/october-9th-is-officially-indigenous-peoples-day-in-minnesota/

As I understand, governor's proclamation is not enough to make it a legal state holiday.

**Review by arkid15r — APPROVED** [review]
LGTM

Please make sure the [CA Columbus Day PR](https://github.com/vacanza/python-holidays/pull/1500) doesn't become dangling due to this overhaul.

**Comment by KJhellico** [comment]
This PR includes changes from #1500 (remove Columbus Day for CA).

**Review by arkid15r — APPROVED** [review]
👍

**Pull request merged**